.github/workflows/release.yml

```diff
@@ -29,13 +29,31 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
+          ref: main
           fetch-depth: 0
           submodules: true
-          persist-credentials: false
+          persist-credentials: true
 
       - uses: astral-sh/setup-uv@v6
         with:
           enable-cache: true
+        
+      - name: Bump project version to match tag
+        id: automatic-version
+        run: |
+          set -Eeuxo pipefail
+          version="${GITHUB_REF#refs/tags/v}"
+          echo ">> uv version $version"
```

Comment:
nitpick: why is this needed? You set -eux above, so all commands will get printed anyway... and GH Actions ALSO will print the commands. So you will have the same thing stated 3x in logs.